doc/time-series.rst

```diff
@@ -150,17 +155,41 @@ For example, we can downsample our dataset from hourly to 6-hourly:
 
 .. ipython:: python
 
-    ds.resample('6H', dim='time', how='mean')
+    ds.resample(time='6H')
+
+This will create a specialized ``Resample`` object which saves information
+necessary for resampling. All of the reduction methods which work with
+``Resample`` objects can also be used for resampling:
+
+.. ipython:: python
+
+   ds.resample(time='6H').mean()
 
-Resample also works for upsampling, in which case intervals without any
-values are marked by ``NaN``:
+You can also supply an arbitrary reduction function to aggregate over each
+resampling group:
 
 .. ipython:: python
 
-    ds.resample('30Min', 'time')
+   ds.resample(time='6H').reduce(np.mean)
+
+For upsampling, xarray provides four methods: ``asfreq``, ``ffill``, ``bfill``,
+and ``interpolate``. ``interpolate`` extends ``scipy.interpolate.interp1d`` and
+supports all of its schemes. All of these resampling operations work on both
+Dataset and DataArray objects with an arbitrary number of dimensions.
+
+.. note::
+
+   The ``resample`` api was updated in version 0.10.0 to reflect similar
+   updates in pandas ``resample`` api to be more groupby-like. Older style
+   calls to ``resample`` will still be supported for a short period:
+
+   .. ipython:: python
+
+    ds.resample('6H', dim='time', how='mean')
+
+   Using the deprecated api, resample will also work for upsampling, in which
+   case intervals without any values are marked by ``NaN``:
```

Comment:
This makes it sound like you can't do upsampling with the new API.

I'm not quite sure what the point of this sentence is -- maybe drop it?